File: app/src/main/java/com/example/speedtest_rework/core/getIP/CurrentNetworkInfo.java
package com.example.speedtest_rework.core.getIP;

import com.example.speedtest_rework.network.NetworkResult;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class CurrentNetworkInfo {
    private double selfLat = 0.0;
    private double selfLon = 0.0;
    private String selfIsp = "";
    private String selfIspIp = "";

    public CurrentNetworkInfo(double selfLat, double selfLon, String selfIsp, String selfIspIp) {
        this.selfLat = selfLat;
        this.selfLon = selfLon;
        this.selfIsp = selfIsp;
        this.selfIspIp = selfIspIp;
    }

    public CurrentNetworkInfo() {

    }

    public double getSelfLat() {
        return selfLat;
    }

    public void setSelfLat(double selfLat) {
        this.selfLat = selfLat;
    }

    public double getSelfLon() {
        return selfLon;
    }

    public void setSelfLon(double selfLon) {
        this.selfLon = selfLon;
    }

    public String getSelfIsp() {
        return selfIsp;
    }

    public void setSelfIsp(String selfIsp) {
        this.selfIsp = selfIsp;
    }

    public String getSelfIspIp() {
        return selfIspIp;
    }

    public void setSelfIspIp(String selfIspIp) {
        this.selfIspIp = selfIspIp;
    }

    public CurrentNetworkInfo getCurrentNetWorkInfo() {
        try {
            URL url = new URL("https://www.speedtest.net/speedtest-config.php");
            HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
            urlConnection.setConnectTimeout(5000);
            urlConnection.setReadTimeout(5000);
            int code = urlConnection.getResponseCode();

            if (code == 200) {
                BufferedReader br = new BufferedReader(
                        new InputStreamReader(
                                urlConnection.getInputStream()));
                String line;
                while ((line = br.readLine()) != null) {
                    if (!line.contains("isp=")) {
                        continue;
                    }
                    this.selfLat = Double.parseDouble(line.split("lat=\"")[1].split(" ")[0].replace("\"", ""));
                    this.selfLon = Double.parseDouble(line.split("lon=\"")[1].split(" ")[0].replace("\"", ""));
                    this.selfIsp = line.split("isp=\"")[1].split(" ")[0].replace("\"", "");
                    this.selfIspIp = line.split("ip=\"")[1].split(" ")[0].replace("\"", "");
                }
                br.close();

            }

        } catch (Exception ex) {
            ex.printStackTrace();
        }
        return this;
    }

    @Override
    public String toString() {
        return "CurrentNetworkInfo{" +
                "selfLat=" + selfLat +
                ", selfLon=" + selfLon +
                ", selfIsp='" + selfIsp + '\'' +
                ", selfIspIp='" + selfIspIp + '\'' +
                '}';
    }
}

File: app/src/main/java/com/example/speedtest_rework/core/getIP/CurrentNetworkInfo.kt
package com.example.speedtest_rework.core.getIP

import android.util.Log
import java.io.BufferedReader
import java.io.InputStreamReader
import java.net.URL
import javax.net.ssl.HttpsURLConnection
import javax.net.ssl.SSLSocketFactory

class CurrentNetworkInfo {
    var selfLat = 0.0
    var selfLon = 0.0
    var selfIsp = ""
    var selfIspIp = ""

    val currentNetWorkInfo: CurrentNetworkInfo
        get() {
            try {
                val url = URL("https://speedtest.net/speedtest-config.php")
                val urlConnection = url.openConnection() as HttpsURLConnection
//                urlConnection.sslSocketFactory = SSLCertificateSocketFactory.getInsecure(0, null)
                urlConnection.sslSocketFactory = SSLSocketFactory.getDefault() as SSLSocketFactory

                urlConnection.setRequestProperty("Cache-Control", "no-cache")
                urlConnection.setRequestProperty("Expires", "-1")
                urlConnection.setRequestProperty("Pragma", "no-cache")
                urlConnection.setRequestProperty("Connection", "Keep-Alive")
                urlConnection.connectTimeout = 5000
                urlConnection.readTimeout = 5000
                val code = urlConnection.responseCode
                if (code == 200) {
                    val br = BufferedReader(
                        InputStreamReader(
                            urlConnection.inputStream
                        )
                    )

                    br.use { r ->
                        if (r != null) {
                            r.lineSequence().forEach {
                                if (it != null) {
                                    if (!it.contains("isp=")) {
                                        return@forEach
                                    }
                                    selfLat =
                                        it.split("lat=\"".toRegex()).dropLastWhile { it.isEmpty() }
                                            .toTypedArray()[1].split(" ".toRegex())
                                            .dropLastWhile { it.isEmpty() }
                                            .toTypedArray()[0].replace("\"", "").toDouble()
                                    selfLon =
                                        it.split("lon=\"".toRegex()).dropLastWhile { it.isEmpty() }
                                            .toTypedArray()[1].split(" ".toRegex())
                                            .dropLastWhile { it.isEmpty() }
                                            .toTypedArray()[0].replace("\"", "").toDouble()
                                    selfIsp =
                                        it.split("isp=\"".toRegex()).dropLastWhile { it.isEmpty() }
                                            .toTypedArray()[1].split(" ".toRegex())
                                            .dropLastWhile { it.isEmpty() }
                                            .toTypedArray()[0].replace("\"", "")
                                    selfIspIp =
                                        it.split("ip=\"".toRegex()).dropLastWhile { it.isEmpty() }
                                            .toTypedArray()[1].split(" ".toRegex())
                                            .dropLastWhile { it.isEmpty() }
                                            .toTypedArray()[0].replace("\"", "")
                                }

                            }
                        }
                    }
                    br.close()
                }
            } catch (ex: Exception) {
                ex.printStackTrace()
            }
            return this
        }

    override fun toString(): String {
        return "CurrentNetworkInfo{" + "selfLat=" + selfLat + ", selfLon=" + selfLon + ", selfIsp='" + selfIsp + '\'' + ", selfIspIp='" + selfIspIp + '\'' + '}'
    }
}